client/js/jsonedit.js

```diff
@@ -1475,16 +1483,59 @@ function jeSetAndSelect(replaceBy, insideString) {
   else
     var jsonString = JSON.stringify(jeStateNow, null, '  ');
   const startIndex = jsonString.indexOf(insideString ? '###SELECT ME###' : '"###SELECT ME###"');
-  let length = jsonString.length-15-(insideString ? 0 : 2);
-
+  const replaceByString = JSON.stringify(replaceBy);
   if(insideString)
-    jsonString = jsonString.replace(/###SELECT ME###/, JSON.stringify(replaceBy).substr(1, JSON.stringify(replaceBy).length-2));
+    jsonString = jsonString.replace(/###SELECT ME###/, replaceByString.substr(1, replaceByString.length-2));
   else
-    jsonString = jsonString.replace(/"###SELECT ME###"/, JSON.stringify(replaceBy));
+    jsonString = jsonString.replace(/"###SELECT ME###"/, replaceByString);
 
   jeSet(jsonString);
-  const quote = typeof replaceBy == 'string' && !insideString ? 1 : 0;
-  jeSelect(startIndex + quote, startIndex+jsonString.length-length - quote, true);
+
+  let bracket = 0;
+  let wavybracket = 0;
+  let begin = 0;
+  let end = 0;
+  if(replaceBy !== undefined){ 
+    const quoteArray = getAllIndexes(replaceByString,'\"');
+    if(quoteArray.length > 2) { // adjusts selection area to inside last set of quotes
+      begin = quoteArray[2] + 1;
+      end = quoteArray[3];
```

Comment:
What case is this designed to handle? Can there ever be more than two sets of quotes? Can there ever be an odd number of quotes?

Reply:
I don't think we will ever have an odd number of quotes.  This is designed to handle things like. `"classes": ""`, which has 1 sets of quotes AND things like `"dropTarget": {"type":"card"}`, which has 2 sets of quotes.  In the first one it will place the cursor where I am placing an x  `"classes": "x"`, in the second one it will place the selection over the entire word `card` `"dropTarget": {"type":"card"}`